nlpnet/network.pyx

```diff
@@ -398,7 +396,7 @@ Output size: %d
         # s = Sum_i(A_tags[i-1],tags[i] + ftheta_i,i), i < len(sentence)   (12)
         correct_path_score = 0
         last_tag = self.output_size
-        for tag, net_scores in izip(tags, scores):
```

Comment:
zip might be a lazy iterator in python 3, but it might not be in python 2 and this might generate some memory bottlenecks, yo could do a `try from itertools import izip as zip` along the lines of cpickle.